### ISSUE TITLE
Get correct `Data` value from `OneSignal.getDeviceState().pushToken` which is a `String`

### ISSUE THREAD

**Issue by JayMehta97** [OneSignal/OneSignal-iOS-SDK#1069]
**Description:**

We are adding another 3rd party SDK for chat which provides PN for chat specific features. This SDK also requires `deviceToken` to setup remote push notifications. So in our current users device we have already taken remote notification access using OneSignal SDK. How do I give `deviceToken` to the new SDK using `OneSignal.getDeviceState().pushToken`? Can I convert the pushToken (String) to correct `deviceToken` (Data)?

**Environment**

iOS version - 14.8.1
OneSignal SDK version - 3.5.3
Using OneSignal SDK via SPM

**Comment by JayMehta97** [comment]
I tried with that method but the [application:didRegisterForRemoteNotificationsWithDeviceToken:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622958-application?language=objc) is not getting called if I just add the [Oneginal SDK](https://github.com/OneSignal/OneSignal-XCFramework) using SPM. If I remove that `didRegisterForRemoteNotificationsWithDeviceToken` is being called but adding it back again doesn't call this method.

**Comment by emawby** [comment]
Odd that method is called for me when using the XCFramework repo and Swift Package Manager. Are you able to link an example project that reproduces the issue?

**Comment by Sml995** [comment]
Hi, we're facing the same issue here with SwiftUI

TLDR;
I suspect this to be relative to swizzling issues addressed with #1091
Do you know when this will be released ?

There's an easy way to reproduce the bug:
- Start a fresh SwiftUI project
- Add [OneSignal-XCFramework](https://github.com/OneSignal/OneSignal-XCFramework) via SPM - v3.11.1
- Add Push notifications capability
- Adjust your root `App` struct as following:
```swift
@main
struct Test_Intercom_OneSignalApp: App {
    
    @UIApplicationDelegateAdaptor(AppDelegate.self) var appDelegate

    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}
```

- Add a file named `AppDelegate.swift` to your main target with the following content:
```swift
import SwiftUI
import OneSignal
import Foundation

class AppDelegate: NSObject, UIApplicationDelegate {
        
    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
        
        //MARK: OneSignal initialization
        OneSignal.setLogLevel(.LL_DEBUG, visualLevel: .LL_NONE)
        OneSignal.initWithLaunchOptions(launchOptions)
        OneSignal.setAppId("your_onesignal_app_id")
        
        OneSignal.setNotificationOpenedHandler( { openedEvent in
            //
        })
        
        OneSignal.promptForPushNotifications(userResponse: { accepted in
            //
        })
        
        return true
        
    }
    
    func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
        //
    }
    
    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
        //
    }
   
}
```

- Place a breakpoint within `didFailToRegisterForRemoteNotificationsWithError` & `didRegisterForRemoteNotificationsWithDeviceToken` and run the app

**Comment by jkasten2** [comment]
@Sml995 Thanks for the details and digging in. I believe you are correct that PR #1091 will fix the issue with your AppDelegate `didFailToRegisterForRemoteNotificationsWithError` and `didRegisterForRemoteNotificationsWithDeviceToken` not firing. We will reply back when we have a release available with this fix.

**Comment by Sml995** [comment]
Getting back on this after [release v3.11.2](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/3.11.2)
Everything is running as expected now :)
